sefaria/model/topic.py

```diff
@@ -438,26 +438,36 @@ def contents(self, **kwargs):
     
     # A person may have an era, a generation, or a specific birth and death years, which each may be approximate.
     # They may also have none of these...
-    def most_accurate_time_period(self) -> Optional[TimePeriod]:
+    def _most_accurate_period(self, obj) -> Optional[LifePeriod]:
         if self.get_property("birthYear") and self.get_property("deathYear"):
-            return TimePeriod({
+            return obj({
                 "start": self.get_property("birthYear"),
                 "startIsApprox": self.get_property("birthYearIsApprox", False),
                 "end": self.get_property("deathYear"),
                 "endIsApprox": self.get_property("deathYearIsApprox", False)
             })
         elif self.get_property("birthYear") and self.get_property("era", "CO"):
-            return TimePeriod({
+            return obj({
                 "start": self.get_property("birthYear"),
                 "startIsApprox": self.get_property("birthYearIsApprox", False),
             })
+        elif self.get_property("deathYear"):
+            return obj({
+                "end": self.get_property("deathYear"),
+                "endIsApprox": self.get_property("deathYearIsApprox", False)
+            })
         elif self.get_property("generation"):
-            return TimePeriod().load({"symbol": self.get_property("generation")})
+            return obj().load({"symbol": self.get_property("generation")})
         elif self.get_property("era"):
-            return TimePeriod().load({"symbol": self.get_property("era")})
+            return obj().load({"symbol": self.get_property("era")})
         else:
             return None
 
+    def most_accurate_time_period(self):
```

Comment:
Can you add documentation for these two functions. They're obviously simple, but I don't think it will be obvious when to use each one.